modules/kodiDB.py

```diff
@@ -198,28 +198,30 @@ def createMusicDB():
         con.close()  
 
 def writeIntoSongTable (pstrSongTitle, songID, pstrArtistName, pstrAlbumName, albumID, path, pathID, purlLink, roleID, artistID, songArtistRel, delSong):
-    selectQuery = ("SELECT id FROM songs WHERE songID=? AND artistID=? AND albumID=?;")
-    selectArgs = (songID, artistID, albumID,)
-    insertQuery = ("INSERT INTO songs (strSongTitle, songID, strArtistName, strAlbumName, albumID, strPath, pathID, strURL, roleID, artistID, songArtistRel, delSong) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);")
-    insertArgs = (pstrSongTitle, songID, pstrArtistName, pstrAlbumName, albumID, path, pathID, purlLink, roleID, artistID, songArtistRel, delSong,)
+    selectQuery = "SELECT id FROM songs WHERE songID='{}' AND artistID='{}' AND albumID='{}';"
+    selectArgs =  (songID, artistID, albumID)
+    insertQuery = "INSERT INTO songs (strSongTitle, songID, strArtistName, strAlbumName, albumID, strPath, pathID, strURL, roleID, artistID, songArtistRel, delSong) VALUES ('{}', '{}', '{}', '{}', '{}', '{}', '{}', '{}', '{}', '{}', '{}', '{}');"
+    path = path if DATABASE_MYSQL == "false" else path.replace('\\', '\\\\')
+    insertArgs =  (pstrSongTitle.replace("'","''"), songID, pstrArtistName, pstrAlbumName, albumID, path, pathID, purlLink, roleID, artistID, songArtistRel, delSong)

```

Comment:
In welchem Fall wird das (pstrSongTitle.replace("'","''")) benötig?

Reply:
Bei bei SQL war es nötig. MySQL störte es nicht.
Wenn es nicht da ist und du hast z.B ein Titel 'It´s my Live' kann cursor.execute
das nicht verarbeiten. Der liest dann 'It' und s my live erzeugt einen Fehler.